### ISSUE TITLE
How to visualize results of test model?

### ISSUE THREAD

**Issue by Latha-13** [open-mmlab/mmdetection3d#604]
Hi

I currently trained and tested the kitti dataset the results are generated as pred.obj and points.obj files .
Now, I have to visualize the 3d bounding box on the 2d image of the tested dataset.
I used "python tools/misc/browse_dataset.py configs/mvxnet/dv_mvx-fpn_second_secfpn_adamw_2x8_80e_kitti-3d-3class.py --task multi_modality-det --output-dir ${OUTPUT_DIR} --online " to visualize but the visualized dataset are trainned dataset. How can I visualize the 3d bbox on 2d image of test dataset?

Thank you

**Comment by Wuziyi616** [comment]
Simple change [this line](https://github.com/open-mmlab/mmdetection3d/blob/master/configs/mvxnet/dv_mvx-fpn_second_secfpn_adamw_2x8_80e_kitti-3d-3class.py#L212) from `kitti_infos_train.pkl` to `kitti_infos_val.pkl`, and we will be able to load and visualize val data. The test data has no GT so we can't visualize them.

**Comment by Latha-13** [comment]
Is there any google collab tutorials for mmdetection3d?
for training and testing the datasets with minimum numbers.

**Comment by Wuziyi616** [comment]
Most of the dataset doesn't provide GT for test set in order to avoid people "overfitting" test data. This includes ScanNet, KITTI, Waymo and others I think. People sometimes also report val results in their papers so I think it's also helpful to visualize the val data. After all, train, val and test should be similar.

About google colab, we haven't discussed this option. Maybe I can raise this in this week's group meeting.

**Comment by Latha-13** [comment]
Hi @Wuziyi616 
1. can you say what are the information present in the obj file?
2.  Is there any posibility to convert the obj files genrated into json or csv files? such that i can use my code to  visualize the 3d bbox on image.

**Comment by Wuziyi616** [comment]
For the obj files, you can simply download MeshLab and easily visualize it. You may need to convert the format of those obj files. See [here](https://github.com/open-mmlab/mmdetection3d/blob/master/mmdet3d/core/visualizer/show_result.py#L10) and [here](https://github.com/open-mmlab/mmdetection3d/blob/master/mmdet3d/core/visualizer/show_result.py#L32) for more details. I think you can try with trimesh to load and save them?

**Comment by Latha-13** [comment]
Hi
Can you said explain in what are the contents are present in the obj file can we read those file?
and also what the contents present in the pkl file?
for eg. xcentre, y centre, l, w, h some thing like that.....

**Comment by Wuziyi616** [comment]
The obj files are just bboxes, the vertices and edges presented in obj format which is commonly used. The pkl files are list of prediction results `List[dict(3d_boxes=xxx, 3d_labels=xxx, 3d_labels=xxx)]`.

**Comment by Latha-13** [comment]
Thank you so much for you response it helped me in understanding.

**Comment by Wuziyi616** [comment]
Glad it helps :)
